evap/staff/importers.py

```diff
@@ -417,6 +470,93 @@ def process(cls, excel_content, test_run):
                 raise
 
 
+class PersonImporter:
+    def __init__(self, import_type, course):
+        self.import_type = import_type
+        self.course = course
+        self.user_list = []
+        self.users_to_add = []
+
+        self.success_messages = []
+        self.warnings = defaultdict(list)
+        self.errors = []
+
+    def handle_already_related(self):
+        already_related = []
+
+        if self.import_type == "participant":
+            usernames = [user.username for user in self.user_list]
+            already_related = self.course.participants.filter(username__in=usernames).all()
+        else:  # import_type == "contributor"
+            already_related = Contribution.objects.filter(course=self.course, contributor__in=self.user_list).all()
+
+        if already_related:
+            if self.import_type == "participant":
+                msg = _("The following {} user(s) are already course participants in course {}:").format(len(already_related), self.course.name)
+            else:  # import_type == "contributor"
+                msg = _("The following {} user(s) are already contributing to course {}:").format(len(already_related), self.course.name)
+            msg += create_user_list_string_for_message(already_related)
+
+        # since the user profiles are not necessarily saved to the database, they are not guaranteed to have a pk yet which
+        # makes anything relying on hashes unusable here (for a faster list difference)
+        self.users_to_add = list(filter(lambda x: x not in already_related, self.user_list))
```

Comment:
in other places we use list comprehensions (`[user for user in...`) for exactly the same purpose, so i'd suggest using that here as well.